addons/web/static/tests/webclient/actions/push_state.test.js

```diff
@@ -0,0 +1,602 @@
+        ],
+    },
+]);
+
+class Partner extends models.Model {
+    name = fields.Char();
+    foo = fields.Char();
+    parent_id = fields.Many2one({ relation: "partner" });
+    child_ids = fields.One2many({ relation: "partner", relation_field: "parent_id" });
+
+    _records = [
+        { id: 1, name: "First record", foo: "yop" },
+        { id: 2, name: "Second record", foo: "blip" },
+        { id: 3, name: "Third record", foo: "gnap" },
+        { id: 4, name: "Fourth record", foo: "plop" },
+        { id: 5, name: "Fifth record", foo: "zoup" },
+    ];
+    _views = {
+        "kanban,1": `
+            <kanban>
+                <templates>
+                    <t t-name="kanban-box">
+                        <div><field name="foo"/></div>
+                    </t>
+                </templates>
+            </kanban>
+        `,
+        list: `<tree><field name="foo"/></tree>`,
+        "list,2": `<tree limit="3"><field name="foo"/></tree>`,
+        form: `
+            <form>
+                <header>
+                    <button name="object" string="Call method" type="object"/>
+                    <button name="4" string="Execute action" type="action"/>
+                </header>
+                <group>
+                    <field name="display_name"/>
+                    <field name="foo"/>
+                </group>
+            </form>
+        `,
+        search: `<search><field name="foo" string="Foo"/></search>`,
+    };
+}
+
+class Pony extends models.Model {
+    name = fields.Char();
+
+    _records = [
+        { id: 4, name: "Twilight Sparkle" },
+        { id: 6, name: "Applejack" },
+        { id: 9, name: "Fluttershy" },
+    ];
+    _views = {
+        list: `<tree><field name="name"/></tree>`,
+        form: `<form><field name="name"/></form>`,
+        search: `<search/>`,
+    };
+}
+
+defineModels([Partner, Pony]);
+
+class TestClientAction extends Component {
+    static template = xml`
+        <div class="test_client_action">
+            ClientAction_<t t-esc="props.action.params?.description"/>
+        </div>
+    `;
+    static props = ["*"];
+
+    setup() {
+        onMounted(() => {
+            this.env.config.setDisplayName(`Client action ${this.props.action.id}`);
+        });
+    }
+}
+
+onRpc("has_group", () => true);
+
+beforeEach(() => {
+    actionRegistry.add("__test__client__action__", TestClientAction);
+    patchWithCleanup(browser.location, {
+        origin: "http://example.com",
+    });
+    redirect("/odoo");
+    startRouter();
+});
+
+/**
+ * @param {{ env: import("@web/env").OdooEnv }} [options]
+ */
+async function mountWebClient(options) {
+    await mountWithCleanup(WebClient, options);
+    // Wait for visual changes caused by a potential loadState
+    await animationFrame();
+    // wait for BlankComponent
+    await animationFrame();
+    // wait for the regular rendering
+    await animationFrame();
```

Comment:
I did not need such an helper in the other suites, are you sure you need it?

Reply:
Indeed, it's not needed for these tests but needed for load state